btree/btree.c

```diff
@@ -3429,21 +3429,19 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 		if (cookie_is_valid(tree, &bop->bo_rec.r_key.k_cookie) &&
 		    !node_isoverflow(oi->i_cookie_node))
 			return P_LOCK;
-		else
-			return P_SETUP;
-	case P_SETUP: {
+		/** Fall through if cookie is invalid or node overflow. */
+	case P_LOCKALL:
+		if (bop->bo_flags & BOF_LOCKALL)
```

Comment:
we should have assert instead of if (bop->bo_flags & BOF_LOCKALL), because we should reach P_LOCKALL only if BOF_LOCKALL flag is set
and we should not have fall-through from BOF_LOCKALL, else there needs to transitions from P_LOCKALL-->P_NEXTDOWN

Reply:
I can do that too. In that case I need to avoid falling from P_COOKIE to P_LOCKALL and jump from P_COOKIE to P_SETUP in case of cookie is invalid.

Reply:
Yes.